packages/schema/src/utils/pkg-utils.ts

```diff
@@ -37,6 +37,32 @@ export function findUp<e extends boolean = false>(names: string[], cwd: string =
     return findUp(names, up, multiple, result);
 }
 
+/**
+ * A function that finds a file within the node_modules directory by the provided name. 
+ * It first searches for the node_modules directory using the findUp function, and if not found, returns undefined. 
+ * It then splits the provided name into folder and file parts. 
+ * It finds the location of the folder in the node_modules directory by checking for its existence. 
+ * If the folder is not found, it returns undefined. 
+ * It then constructs the file location by joining the folder location and the file name with a '.zmodel' extension if the file does not already have an extension. 
+ * Returns the file location if it exists, otherwise returns undefined.
+ * @author Jonathan Stevens (TGTGamer)
+ *
+ * @export
+ * @param name A string representing the name of the file to find in the node_modules folder
+ * @returns Path to a specific file within the node_modules directory
+ */
+export function findNodeModulesFile(name: string, cwd: string = process.cwd()) {
```

Comment:
Do you think we can actually use `require.resolve` to resolve zmodel? Like `require.resolve('@myorg/mypkg/foo.zmodel')` or `require.resolve('pkg/foo.zmodel')`.

I believe `require.resolve` handles recursive upward look-up internally. Maybe it gets us the same behavior?